torchgeo/datasets/l7irish.py

```diff
@@ -0,0 +1,251 @@
+     * Clear
+     * Thin Cloud
+     * Cloud
+
+     If you use this dataset in your research, please cite the following papers:
+
+     * https://doi.org/10.5066/F7XD0ZWC
+     * https://doi.org/10.1109/TGRS.2011.2164087
+     """
+    
+    # naming... [need to change]
+    filename_regex = r"""
+        ^L
+        (?P<sensor>[COTEM])
+        (?P<satellite>\d{2})
+        _(?P<processing_correction_level>[A-Z0-9]{4})
+        _(?P<wrs_path>\d{3})
+        (?P<wrs_row>\d{3})
+        _(?P<date>\d{8})
+        _(?P<processing_date>\d{8})
+        _(?P<collection_number>\d{2})
+        _(?P<collection_category>[A-Z0-9]{2})
+        _(?P<band>[A-Z0-9_]+)
+        \.
+    """
+
+    tarfile_glob = "*.tar.gz"
+
+    url = "https://huggingface.co/datasets/torchgeo/l7irish/resolve/main/{}.tar.gz"
+    
+    md5s = [("austral", "dbb6b5628f50861b9b89f548d25a925f"),
+           ("boreal", "cecc72de09aacde4c4f8d7f0cf0d3f6f"),
+           ("mid_latitude_north", "0f8382ca6554fb7cf9aff42226a14f9d"),
+           ("mid_latitude_south", "b17cf6d023f752c533211fdb742f296b"),
+           ("polar_north", "73923dcaf1b9b79bad82de1aa0740d1e"),
+           ("polar_south", "3bc9f4c6f8955b10b4d55d23e0ab2da7"),
+           ("subtropical_north", "f8f039970256902e6e9ebd6747589294"),
+           ("subtropical_south", "8346d73a983396c5d41b577c3a94bc26"),
+           ("tropical", "abe19b22b5d031e6b609cc7207706c3d")
+           ]
+        
+    classes = ["Fill", "Cloud Shadow", "Clear", "Thin Cloud", "Cloud"]
+    
+    cmap = {
+        0: (0, 0, 0),
+        64: (64, 64, 64),
+        128: (128, 128, 128),
+        192: (192, 192, 192),
+        255: (255, 255, 255),
+    }
+
+    def __init__(
+        self, root: str = "data", download: bool = False, checksum: bool = False
+    ) -> None:
+        
+        """Initialize a new Landsat 7 Cloud Cover Assessment Validation dataset instance.
+
+            Args:
+                root: root directory where dataset can be found
+                transforms: a function/transform that takes input sample and its target as
+                    entry and returns a transformed version
+                cache: if True, cache file handle to speed up repeated sampling
+                download: if True, download dataset and store it in the root directory
+                checksum: if True, check the MD5 of the downloaded files (may be slow)
+
+            Raises:
+                RuntimeError: if ``download=False`` and data is not found, or checksums
+                    don't match
+            """
+        self.root = root
+        self.download = download
+        self.checksum = checksum
+    
+        for i in list(self.cmap.keys()):
+            self.cmap[np.ceil(i/64).astype(int)] = self.cmap.pop(i)
+
+        lc_colors = np.zeros((max(self.cmap.keys()) + 1, 3))
+        lc_colors[list(self.cmap.keys())] = list(self.cmap.values())
+        lc_colors = lc_colors / 255
+        _lc_cmap = ListedColormap(lc_colors)
```

Comment:
What does this code do and why is it needed? "I just copied it from CDL" isn't a good answer 😛